common/membership/ringpop/hostinfo.go

```diff
@@ -25,19 +25,31 @@
 // Package ringpop provides a service-based membership monitor
 package ringpop
 
+import (
+	"fmt"
+
+	"github.com/dgryski/go-farm"
+	rpmembership "github.com/temporalio/ringpop-go/membership"
+
+	"go.temporal.io/server/common/membership"
+)
+
+// hostInfo represents a host in a ring.
 type hostInfo struct {
-	addr   string // ip:port
-	labels map[string]string
+	addr           string            // ip:port
+	labels         map[string]string // this map is shared with ringpop
+	labelsChecksum uint64            // cache this for quick comparison
 }
 
+var _ rpmembership.Member = (*hostInfo)(nil)
```

Comment:
hostInfo objects look to be immutable, so why bother passing around pointers to them?